### PR TITLE
Update detray version (v0.11.0 -> v0.12.0) and replace track parameter classes

### PR DESCRIPTION
**Pull request by beomki-yeo**
This PR updates the detray version and removes traccc track parameter classes to utilize detray track parameter classes

### CODE REVIEW

File: device/common/include/traccc/seeding/device/impl/estimate_track_params.ipp [hunk 1]
```diff
@@ -35,11 +35,12 @@ inline void estimate_track_params(
     vector3 bfield = {0, 0, 2};
 
     const seed& this_seed = seeds_device.at(globalIndex);
-    bound_vector& param = params_device[globalIndex].vector();
+    // bound_vector param = params_device[globalIndex].vector();
```
Comment (krasznaa) [inline device/common/include/traccc/seeding/device/impl/estimate_track_params.ipp:38]: I guess this line could be removed.

**Review by krasznaa — COMMENTED** [review]
It looks good, we just need to update it to the latest `main` version...

File: device/common/include/traccc/seeding/device/impl/estimate_track_params.ipp (hunk 1 above)
Comment (beomki-yeo) [reply device/common/include/traccc/seeding/device/impl/estimate_track_params.ipp]: Yes

**Review by krasznaa — APPROVED** [review]
Let's do it! :smile:

**Pull request merged**